com.unity.ml-agents/Runtime/Analytics/TrainingAnalytics.cs

```diff
@@ -211,10 +211,14 @@ public static void TrainingBehaviorInitialized(TrainingBehaviorInitializedEvent
                 return;
             }
 
-            // Hash the behavior name so that there's no concern about PII or "secret" data being leaked.
             tbiEvent.TrainingSessionGuid = s_TrainingSessionGuid.ToString();
-            tbiEvent.BehaviorName = AnalyticsUtils.Hash(tbiEvent.BehaviorName);
 
+            if(tbiEvent.Config.Length == 0 || tbiEvent.BehaviorName.Length != 64) {
+              // Hash the behavior name if the message version is from an older version of ml-agents that doesn't do trainer-side hashing.
+              // We'll also, for extra safety, verify that the BehaviorName is the size of the expected SHA256 hash.
+              // Context: The config field was added at the same time as trainer side hashing, so messages including it should already be hashed.
+              tbiEvent.BehaviorName = AnalyticsUtils.Hash(k_VendorKey, tbiEvent.BehaviorName);
+            }
```

Comment:
Besides adding python unit tests, it would be nice to also update the C# training analytics test to make sure it can properly handle this

Reply:
+1